### ISSUE TITLE
Steam Overlay continues to plague me even with it disabled globally and per-game

### ISSUE THREAD

**Issue by colemickens** [ValveSoftware/steam-for-linux#10093]
#### Your system information

* Steam client version (build number or date): `Steam Version:  1695679261`
* Distribution (e.g. Ubuntu): `NixOS`
* Opted into Steam client beta?: `Yes`
* Have you checked for system updates?: `n/a, I upgrade multiple times a day, it's a problem`
* Steam Logs: (I'm wary of just uploading a mass of logs without combing through them, if they're needed I can)
* GPU: `AMD 6700S`

#### Please describe your issue in as much detail as possible:

I have the Steam Overlay disabled everywhere I can think of:
* globally:
   ![screenshot-1695853583](https://github.com/ValveSoftware/steam-for-linux/assets/327028/8c33b00f-c988-4be0-be50-10fc14e92c2f)
* per-game:
   ![screenshot-1695853605](https://github.com/ValveSoftware/steam-for-linux/assets/327028/56a7c680-1747-4dd8-ab8f-7b85a9056dff)

and yet, if I make the mistake of hitting Shift and Tab (extremely common when playing ... lots of games):

![screenshot-1695853662](https://github.com/ValveSoftware/steam-for-linux/assets/327028/b352bf43-db74-463e-a418-8c14cece48ce)

**Comment by colemickens** [comment]
The only thought I have is that it's something to do with Vulkan, the steam overlay vulkan layer, and the fact that I run sway with `WLR_RENDERER=vulkan`? But that's just a completely random guess.

**Comment by YellowOnion** [comment]
Yeah Steam is stupid and doesn't actually disable the overlay, I spent ages debugging an issue related to it.

Try launch the game like `LD_PRELOAD='' %command%`, Steam manually injects the library, this overrides the injection.
